googledataprocauthenticator/google.py

```diff
@@ -0,0 +1,319 @@
+
+def load_json_input(result):
+    """Load json data from the file."""
+    jsondata = None
+    try:
+        jsondata = json.loads(result)
+    except:
+        raise
+    return jsondata
+
+def list_accounts_pairs(credentialed_accounts, default_credentials_configured):
+    """Reformats all of user's credentialed accounts to populate google_credentials_widget
+    dropdown's options. 
+
+    Args:
+        credentialed_accounts (str): user credentialed account to return credentials for
+        default_credentials_configured (boolean): This is True if google application-default
+        credentials are configured.
+
+    Returns:
+        dict: each key is a str of the users credentialed accounts and it maps to the 
+        same str credentialed account
+    """
+    accounts = credentialed_accounts
+    
+    accounts_dict = {}
+    for account in accounts:
+        accounts_dict[account] = account
+    if default_credentials_configured:
+        accounts_dict['default-credentials'] = 'default-credentials'
+    return accounts_dict
+
+
+def list_credentialed_accounts():
+    """Load all of user's credentialed accounts with ``gcloud auth list`` command.
+
+    Returns:
+        Sequence[str]: each key is a str of one of the users credentialed accounts
+
+    Raises:
+        sparkmagic.livyclientlib.BadUserConfigurationException: if account is not set or user
+        needs to run gcloud auth login or if gcloud is not installed.
+    """
+    accounts_json = ""
+    if os.name == "nt":
+        command = _CLOUD_SDK_WINDOWS_COMMAND
+    else:
+        command = _CLOUD_SDK_POSIX_COMMAND
+    try:
+        command = (command,) + _CLOUD_SDK_USER_CREDENTIALED_ACCOUNTS_COMMAND
+        accounts_json = subprocess.check_output(command, stderr=subprocess.STDOUT)
+        all_accounts = load_json_input(accounts_json)
+        credentialed_accounts = set()
+        active_account = None
+        for account in all_accounts:
+            try:
+                _cloud_sdk.get_auth_access_token(account['account'])
+                if account['status'] == 'ACTIVE':
+                    active_account = account['account']
+                credentialed_accounts.add(account['account'])
+            except:
+                pass
+        return credentialed_accounts, active_account
+    except (OSError) as caught_exc:
+        new_exc = BadUserConfigurationException(
+            "Gcloud is not installed. Install the Google Cloud SDK.")
+        raise new_exc
+    except (subprocess.CalledProcessError, IOError) as caught_exc:
+        new_exc = BadUserConfigurationException(
+            "Failed to obtain access token. Run `gcloud auth login` in your command line "\
+            "to authorize gcloud to access the Cloud Platform with Google user credentials to "\
+            "authenticate. Run `gcloud auth application-default login` acquire new user "\
+            "credentials to use for Application Default Credentials."
+        )
```

Comment:
Are you intentionally omitting the text of `caught_exc` because gcloud would have already printed the error? (If so you should add a comment discussing that)